File: app/src/main/java/jp/hazuki/yuzubrowser/webkit/handler/WebSrcImageOpenRightNewTabHandler.java
package jp.hazuki.yuzubrowser.webkit.handler;

import java.lang.ref.WeakReference;

import jp.hazuki.yuzubrowser.webkit.TabType;
import jp.hazuki.yuzubrowser.webkit.WebBrowser;

public class WebSrcImageOpenRightNewTabHandler extends WebSrcImageHandler {
    private final WeakReference<WebBrowser> mReference;

    public WebSrcImageOpenRightNewTabHandler(WebBrowser web) {
        mReference = new WeakReference<>(web);
    }

    @Override
    public void handleUrl(String url) {
        WebBrowser web = mReference.get();
        if (web != null)
            web.openInRightNewTab(url, TabType.WINDOW);
    }
}

File: legacy/src/main/java/jp/hazuki/yuzubrowser/legacy/webkit/handler/WebSrcImageOpenRightNewTabHandler.kt
/*
 * Copyright (C) 2017 Hazuki
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package jp.hazuki.yuzubrowser.legacy.webkit.handler

import jp.hazuki.yuzubrowser.legacy.browser.BrowserController
import jp.hazuki.yuzubrowser.legacy.webkit.TabType
import java.lang.ref.WeakReference

class WebSrcImageOpenRightNewTabHandler(controller: BrowserController) : WebSrcImageHandler() {
    private val mReference: WeakReference<BrowserController> = WeakReference(controller)

    override fun handleUrl(url: String) {
        mReference.get()?.openInRightNewTab(url, TabType.WINDOW)
    }
}
